src/nbl/asset/utils/CIESProfile.cpp

```diff
@@ -0,0 +1,300 @@
+            case OTHER_HALF_SYMMETRIC:
+                return abs(_phi); //! eg. maps (in degress) 181 -> 179 or 359 -> 1
+            case NO_LATERAL_SYMMET: //! plot onto whole (in degress) [0, 360] range
+            {
+                if (_phi < 0)
+                    return _phi + M_TWICE_PI;
+                else
+                    return _phi;
+            }
+            default:
+                assert(false);
+        }
+    };
+
+    const float vAngle = core::degrees(theta), hAngle = core::degrees(wrapPhi(phi));
+
+    assert(vAngle >= 0.0 && vAngle <= 180.0);
+    assert(hAngle >= 0.0 && hAngle <= 360.0);
+
+    if (vAngle > vAngles.back())
+        return 0.0;
+
+    // bilinear interpolation
+    auto lb = [](const core::vector<double>& angles, double angle) -> size_t
+    {
+        assert(!angles.empty());
+        const size_t idx = std::upper_bound(std::begin(angles), std::end(angles), angle) - std::begin(angles);
+        return (size_t)std::max((int64_t)idx - 1, (int64_t)0);
+    };
+
+    auto ub = [](const core::vector<double>& angles, double angle) -> size_t
+    {
+        assert(!angles.empty());
+        const size_t idx = std::upper_bound(std::begin(angles), std::end(angles), angle) - std::begin(angles);
+        return std::min<size_t>(idx, angles.size() - 1);
+    };
+
+    const size_t j0 = lb(vAngles, vAngle);
+    const size_t j1 = ub(vAngles, vAngle);
+    const size_t i0 = symmetry == ISOTROPIC ? 0 : lb(hAngles, hAngle);
+    const size_t i1 = symmetry == ISOTROPIC ? 0 : ub(hAngles, hAngle);
+
+    double uResp = i1 == i0 ? 1.0 : 1.0 / (hAngles[i1] - hAngles[i0]);
+    double vResp = j1 == j0 ? 1.0 : 1.0 / (vAngles[j1] - vAngles[j0]);
+
+    double u = (hAngle - hAngles[i0]) * uResp;
+    double v = (vAngle - vAngles[j0]) * vResp;
+
+    double s0 = getValue(i0, j0) * (1.0 - v) + getValue(i0, j1) * (v);
+    double s1 = getValue(i1, j0) * (1.0 - v) + getValue(i1, j1) * (v);
+
+    return s0 * (1.0 - u) + s1 * u;
+}
+
+inline core::vectorSIMDf CIESProfile::octahdronUVToDir(const float& u, const float& v)
+{
+    core::vectorSIMDf pos = core::vectorSIMDf(2 * (u - 0.5), 2 * (v - 0.5), 0.0);
+    float abs_x = core::abs(pos.x), abs_y = core::abs(pos.y);
+    pos.z = 1.0 - abs_x - abs_y;
+    if (pos.z < 0.0) {
+        pos.x = core::sign(pos.x) * (1.0 - abs_y);
+        pos.y = core::sign(pos.y) * (1.0 - abs_x);
+    }
+
+    return core::normalize(pos);
+}
+
+
+inline std::pair<float, float> CIESProfile::sphericalDirToRadians(const core::vectorSIMDf& dir)
+{
+    const float theta = std::acos(std::clamp<float>(dir.z, -1.f, 1.f));
+    const float phi = std::atan2(dir.y, dir.x);
+
+    return { theta, phi };
+}
+
+// this would not be required if we were to use c++20 on this branch
+template<typename T>
+T atomic_fetch_add(std::atomic<T>* obj, T arg) {
+    T expected = obj->load();
+    while (!std::atomic_compare_exchange_weak(obj, &expected, expected + arg));
+
+    return expected;
+}
```

Comment:
the proper way to parallelize is for each thread/lane to allocate/grab its own temp `T` value and += into that by itself, and when the job completes, add up all the temps together, this is better because:
- no atomic contention
- IEEE754 issues with adding small values to a large accumulator are mitigated